src/compiler/program.ts

```diff
@@ -626,13 +627,30 @@ namespace ts {
                 onlyRecordFailures = !directoryProbablyExists(directory, state.host);
             }
         }
-        return forEach(extensions, tryLoad);

```

Comment:
Could it be `function worker(candidate)` in a closure or would that be bad for perf?